File: clients/client-python/gravitino/api/source_engine_type.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

from enum import Enum


class SourceEngineType(Enum):
    """An enum class containing fileset data operations source engine type that supported."""

    PYSPARK = "pyspark"
    CLOUDML = "cloudml"
    NOTEBOOK = "notebook"
    UNKNOWN = "unknown"

File: api/src/main/java/com/datastrato/gravitino/file/SourceEngineType.java
/*
 * Copyright 2024 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.file;

/** An enum class containing fileset data operations source engine type that supported. */
public enum SourceEngineType {
  SPARK,
  PYSPARK,
  FLINK,
  CLOUDML,
  NOTEBOOK,
  UNKNOWN;
}
